devito/ir/support/basic.py

```diff
@@ -266,7 +280,12 @@ def distance(self, other, findex=None):
                 raise TypeError("Cannot compute distance as `findex` not in `findices`")
         else:
             limit = self.rank
-        return super(IterationInstance, self).distance(other)[:limit]
+        distance = super(IterationInstance, self).distance(other)[:limit]
+        if view is None:
+            return distance
+        else:
+            proj = [i for i, d in zip(distance, self.findices) if d in as_tuple(view)]
```

Comment:
This caught my eye; I had to read it twice to realise it was right.  Having `i` for "distance" and `d` for "index" I think is confusing?  (I know it's 'd' because it's a dimension, but maybe 'dist' instead of 'i')?

Reply:
oooooohhh yes! that's quite horrible, so I'll definitely swap them :)

Reply:
done